packages/trpc/server/routers/viewer/eventTypes/getUserEventGroups.handler.ts

```diff
@@ -59,9 +61,29 @@ export const getUserEventGroups = async ({ ctx, input }: GetByViewerOptions) =>
   const profileProcessor = new ProfilePermissionProcessor();
   const profiles = profileProcessor.processProfiles(eventTypeGroups, teamPermissionsMap);
 
+  const permissionCheckService = new PermissionCheckService();
+
+  const teamPermissionsArray = await Promise.all(
+    filteredEventTypeGroups
+      .map((group) => group.teamId)
+      .filter((teamId): teamId is number => teamId !== null && teamId !== undefined)
```

Comment:
Just more readable  and easily understandable

```
  const teamIdsToCheck = [...new Set(
    filteredEventTypeGroups
      .map(group => group.teamId)
      .filter((id): id is number => id != null)
  )];

  const teamPermissionChecks = teamIdsToCheck.map(async (teamId) => {
    const hasPermission = await permissionCheckService.checkPermission({
      userId: user.id,
      teamId,
      permission: CREATE_EVENT_TYPE_PERMISSION,
      fallbackRoles: [MembershipRole.OWNER, MembershipRole.ADMIN],
    });

    return {
      teamId,
      permissions: { canCreateEventType: hasPermission }
    };
  });

const permissionResults = await Promise.all(teamPermissionChecks);
```

Reply:
applied !